src/client/api/mqtt/sub_milestone_latest.c

```diff
@@ -0,0 +1,79 @@
+
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include "client/api/mqtt/sub_milestone_latest.h"
+#include <stdlib.h>
+#include "client/api/json_utils.h"
+#include "client/network/mqtt/mqtt.h"
+
+void (*milestone_latest_cb)(res_milestone_latest_t *res);
+
+res_milestone_latest_t *res_milestone_latest_new(void){
+  res_milestone_latest_t *res = (res_milestone_latest_t *)malloc(sizeof(res_milestone_latest_t));
+  if (res) {
+    res->is_error = false;
+    res->u.received_milestone_latest = (milestone_latest_t *)malloc(sizeof(milestone_latest_t));
+    return res;
+  }
+  return NULL;
+}
+
+void res_milestone_latest_free(res_milestone_latest_t *res) {
+  if (res) {
+    if (res->is_error) {
+      free(res->u.error);
+    } else {
+      if (res->u.received_milestone_latest) {
+        free(res->u.received_milestone_latest);
+      }
+    }
+    free(res);
+  }
+}
+
+void mqtt_cb_milestone_latest(void *payload) {
+
+  cJSON *json_obj = cJSON_Parse((char *)payload);
+  if (json_obj == NULL) {
+    printf("[%s:%d] OOM\n", __func__, __LINE__);
+    goto end;
+  }
+
+  // Allocates milestone latest response object
+  res_milestone_latest_t *res = res_milestone_latest_new();
+
+  // Parse index
+  if ((json_get_uint32(json_obj, JSON_KEY_INDEX, &(res->u.received_milestone_latest->index))) != 0) {
+    printf("[%s:%d]: parse %s failed\n", __func__, __LINE__, JSON_KEY_INDEX);
+    res->is_error = true;
+    char *error_msg = "Parse index failed";
+    res->u.error = (char*)malloc(strlen(error_msg));
+    strcpy(res->u.error, error_msg);
```

Comment:
it's good that return an error object but it should prevent dynamic allocations. since there is an error message via `printf`, it seems redundant.

Reply:
In this case, 
Option 1 : Inseatd of using res_milestone_latest_t structure
as paramter for callback function, I can directly use milestone_latest_t as the parameter for callback. I used res_milestone_latest_t structure format to keep the http response and mqtt response similar to each other. (Note : But here bool is_error field wont be present, we will pass data to user defined callback only if received response is as expected)

Option 2 :  Change res_milestone_latest_t  to
typedef struct {
  bool is_error;  ///< True if error occured
  milestone_latest_t *received_milestone_latest;  ///< a balance object if is_error is False
} res_milestone_latest_t;

Which do you suggest?

Reply:
Made Changes.